src/rdkafka_telemetry_encode.c

```diff
@@ -34,78 +34,379 @@
 #include "nanopb/pb_decode.h"
 #include "opentelemetry/metrics.pb.h"
 
-typedef union {
-        int32_t intValue;
-        double doubleValue;
-} rd_kafka_telemetry_metric_value_t;
-
-typedef rd_kafka_telemetry_metric_value_t (
-    *rd_kafka_telemetry_metric_value_calculator_t)(rd_kafka_t *);
+#define RDKAFKA_TELEMETRY_NS_TO_MS_FACTOR 1000000
 
 typedef struct {
         opentelemetry_proto_metrics_v1_Metric **metrics;
         size_t count;
 } rd_kafka_telemetry_metrics_repeated_t;
 
 typedef struct {
-        const char *name;
-        const char *description;
-        const char *unit;
-        const rd_bool_t is_int;
-        rd_kafka_telemetry_metric_type_t type;
-        rd_kafka_telemetry_metric_value_calculator_t calculate_value;
-} rd_kafka_telemetry_metric_info_t;
+        opentelemetry_proto_common_v1_KeyValue **key_values;
+        size_t count;
+} rd_kafka_telemetry_key_values_repeated_t;
 
-static rd_kafka_telemetry_metric_value_t
-calculate_connection_creation_total(rd_kafka_t *rk) {
+
+static rd_kafka_telemetry_metric_value_t calculate_connection_creation_total(
+    rd_kafka_t *rk,
+    rd_kafka_broker_t *broker,
+    rd_kafka_telemetry_temporality_type_t temporality) {
         rd_kafka_telemetry_metric_value_t total;
         rd_kafka_broker_t *rkb;
 
         total.intValue = 0;
         TAILQ_FOREACH(rkb, &rk->rk_brokers, rkb_link) {
-                total.intValue += rkb->rkb_c.connects.val;
+                if (temporality ==
+                    RD_KAFKA_TELEMETRY_TEMPORALITY_TYPE_CUMULATIVE)
+                        total.intValue += rkb->rkb_c.connects.val;
+                else
+                        total.intValue += rkb->rkb_c.connects.val -
+                                          rkb->rkb_c_historic.connects;
         }
 
         return total;
 }
 
-static rd_kafka_telemetry_metric_value_t
-calculate_connection_creation_rate(rd_kafka_t *rk) {
+static rd_kafka_telemetry_metric_value_t calculate_connection_creation_rate(
```

Comment:
Should we average this per second as the definition is the connnection creation per second? Otherwise it's same as calculate_connection_creation_total with delta temporality